discord/file.py

```diff
@@ -143,3 +147,59 @@ def close(self) -> None:
         self.fp.close = self._closer
         if self._owner:
             self._closer()
+
+
+class VoiceMessage(File):
+    """A special case of the File class that represents a voice message.
+
+    .. versionadded:: 2.7
+
+    .. note::
+
+        Just like File objects VoiceMessage objects are single use and are not meant to be reused in
+        multiple requests.
+
+    Attributes
+    ----------
+    fp: Union[:class:`os.PathLike`, :class:`io.BufferedIOBase`]
+        A audio file-like object opened in binary mode and read mode
+        or a filename representing a file in the hard drive to
+        open.
+
+        .. note::
+
+            If the file-like object passed is opened via ``open`` then the
+            modes 'rb' should be used.
+
+            To pass binary data, consider usage of ``io.BytesIO``.
+
+    filename: Optional[:class:`str`]
+        The filename to display when uploading to Discord.
+        If this is not given then it defaults to ``fp.name`` or if ``fp`` is
+        a string then the ``filename`` will default to the string given.
+    description: Optional[:class:`str`]
+        The description of a file, used by Discord to display alternative text on images.
+    spoiler: :class:`bool`
+        Whether the attachment is a spoiler.
+    waveform: Optional[:class:`str`]
+        The base64 encoded bytearray representing a sampled waveform. Currently only for voice messages
+    duration_secs: Optional[:class:`float`]
+        The duration of the audio file. Currently only for voice messages
```

Comment:
This is already in the VoiceMessage class so it should be clear. Also, why not just call this `duration`?
```suggestion
        The duration of the voice message.
```

Reply:
That is what it is called in the Discord API